src/components/url-form.tsx

```diff
@@ -0,0 +1,90 @@
+'use client';
+
+import { Form, FormControl, FormField, FormItem } from './ui/form';
+import { Input } from './ui/input';
+import { Button } from './ui/button';
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { cn } from '@/lib/utils';
+import ErrorAlert from './error-alert';
+import { useState } from 'react';
+
+const formSchema = z.object({
+  url: z
+    .string()
+    .min(1, 'Field Required')
+    .regex(
+      /^(https:\/\/)(www\.)?github\.com\/[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+\/?$/,
+      // /^(https:\/\/)(www\.)?github\.com\/[a-zA-Z0-9_-]+\/[a-zA-Z0-9_-]+.*\/?$/, // This regex includes the possibility of having a path & query params after the repo name
+      'Not a valid Github Repo URL'
+    ),
+});
+
+async function fetchRepoIssues(inputUrl: string) {
+  const url = new URL(inputUrl);
+  const [, owner, repo] = url.pathname.split('/');
+
+  return fetch(`https://api.github.com/repos/${owner}/${repo}/issues`);
+}
+
+interface GithubUrlFormProps {
+  className?: string;
+}
+
+export default function GithubUrlForm({ className }: GithubUrlFormProps) {
+  const form = useForm<z.infer<typeof formSchema>>({
+    resolver: zodResolver(formSchema),
+  });
+
+  const [invalidRepoError, setInvalidRepoError] = useState(false);
+
+  async function onSubmit(data: z.infer<typeof formSchema>) {
+    try {
+      const res = await fetchRepoIssues(data.url);
+
+      if (res.status == 404) {
+        setInvalidRepoError(true);
+        return;
+      }
+
+      console.log(await res.json());
+    } catch {
+      console.log('An unexpected error occured, please try again later.');
+    }
+  }
+
+  return (
+    <>
+      <Form {...form}>
+        <form
+          onSubmit={form.handleSubmit(onSubmit)}
+          className={cn('space-y-2', className)}
+        >
+          {(form.formState.errors.url && (
+            <ErrorAlert>{form.formState.errors.url.message}</ErrorAlert>
```

Comment:
ErrorAlert component is nice but lets keep it simple with red text feedback below the input. You can do this by simply passing in {message: "..."} into the various types in the schema. Take a look at the input form example: https://ui.shadcn.com/docs/components/input#form